pkg/deploy/controller/deployment/controller.go

```diff
@@ -144,13 +144,18 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// Generate a deployer pod spec.
 		deployerPod, err := c.makeDeployerPod(deployment)
 		if err != nil {
-			return fatalError(fmt.Sprintf("couldn't make deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
+			nextStatus = deployapi.DeploymentStatusFailed
+			updatedAnnotations[deployapi.DeploymentStatusReasonAnnotation] = err.Error()
+			c.emitDeploymentEvent(deployment, kapi.EventTypeWarning, "FailedDecoding", err.Error())
```

Comment:
"FailedGeneration" might be a more encompassing event code. Presumably the error text would describe the decoding error?